### PR TITLE
replace selenium with webdriverio & extract util functions from selenium test 

### PR DESCRIPTION
**Pull request by mminder**
## Description
- Replace selenium-webdriver with webdriver.io as test script framework
- Extract setup and util functions into separate file to have better overview of actual tests
- Implement recovery use case test

## Motivation and Context
Webdriver.io has some advantages over selenium-webdriver, e.g.:
  * Improved error logging: If a test fails, the log contains the actual lines of test code that failed. This enables faster bugfixing and is just convenient.
  * Better documentation: Webdriver.io docs (https://webdriver.io/docs/gettingstarted) provide extensive documentation such as setup guides, good practices and configuration options. 
  * Flexibility for future testing needs: Webdriver.io can switch between webdriver-protocol and devTools protocol (chrome/puppeteer). It can also be used to write Appium tests, if in the future we decide to automate android/iOS ui tests.
  * API is a bit more concise (e.g. 'browser.$("#registerButton").click();'  vs. 'driver.findElement(By.id("registerButton")).click();' )

## How Has This Been Tested?
local and CI

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: src/frontend/src/test-e2e/selenium.test.ts [hunk 1]
```diff
@@ -55,453 +41,368 @@ const DEMO_APP_URL = "http://localhost:8080/";
...
-    await alert1.accept();
+    const alertText1 = await browser.getAlertText();
     expect(alertText1).toBe(
       "This will remove your current device and you will be logged out"
     );
-    await driver.wait(until.alertIsPresent());
-    const alert2 = driver.switchTo().alert();
-    const alertText2 = await alert2.getText();
-    await alert2.accept();
+    await browser.acceptAlert();
```
Comment (nmattia) [inline src/frontend/src/test-e2e/selenium.test.ts:387]: Shouldn't this be done after we checked the alter text?

File: src/frontend/src/test-e2e/util.ts [hunk 2]
```diff
@@ -0,0 +1,209 @@
...
+  if (outer) {
+    webdriverProcess = await startWebdriver();
+  }
+
+  const browser = await remote({
+    capabilities: {
+      browserName: "chrome",
+      "goog:chromeOptions": {
+        args: [
+          "--no-sandbox",
```
Comment (nmattia) [inline src/frontend/src/test-e2e/util.ts:56]: why disable the sandbox?

**Review by nmattia — COMMENTED** [review]
Very cool! Out of my league here so I'll let @kritzcreek review as well.

File: src/frontend/src/test-e2e/util.ts (hunk 2 above)
Comment (mminder) [reply src/frontend/src/test-e2e/util.ts:56]: I copied that from somewhere but it doesn't seem necessary. Good point, I'll remove it :-)

File: src/frontend/src/test-e2e/selenium.test.ts (hunk 1 above)
Comment (mminder) [reply src/frontend/src/test-e2e/selenium.test.ts:383]: I kept the logic as it was before. There are two alerts in succession and it seems to sufficient to wait for the second one

**Review by kritzcreek — APPROVED** [review]
Looks great! Nice that the async usage has been made to look nicer than the docs would suggest.

**Pull request merged**